content/applications/sales/sales/invoicing/milestone.rst

```diff
@@ -2,32 +2,244 @@
+
+When the order is confirmed, new smart buttons appear at the top of the sales order based on what
+was selected in the :guilabel:`Create on Order` field on the product form.
+
+.. image:: milestone/confirmed-sales-order-milestone.png
+   :align: center
+   :alt: Confirmed sales order with milestone products in the order lines tab.
+
+From the sales order, click the :guilabel:`Milestones` smart button. Doing so reveals a populated
+:guilabel:`Milestones` page. Click :guilabel:`New` to add milestones.
+
+.. image:: milestone/adding-milestones.png
+   :align: center
+   :alt: Adding milestones to a sales order with milestone products.
+
+Enter a :guilabel:`Name` for the milestone. Next, apply it to the corresponding :guilabel:`Sales
+Order Item`. Lastly, assign a :guilabel:`Deadline` to the milestone, if desired.
+
+.. note::
+   The :guilabel:`Quantity` field is not editable.
+
+Repeat that process for all milestone sales order items.
+
+Then, return to the sales order, via the breadcrumbs. From the sales order, click the
+:guilabel:`Tasks` smart button. Doing so reveals a :guilabel:`Tasks` page with a task for each sales
+order item with that option designated in the :guilabel:`Create on Order` field.
+
+.. image:: milestone/tasks-page.png
+   :align: center
+   :alt: Sample tasks page accessed via the smart button from a sales order with milestone products.
+
+To manually assign a configured milestone to a task, click the desired task, which reveals the task
+form. On the task form, select the appropriate milestone to which this task should be connected, in
+the :guilabel:`Milestone` field.
+
+.. image:: milestone/milestone-field-on-task-form.png
+   :align: center
+   :alt: The milestone field on the task form when dealing with milestone products in Odoo Sales.
+
+Repeat this process for all milestone tasks.
+
+With those tasks properly configured, employees will log in their progress as they work on the task,
+in addition to adding any notes related to the task.
+
+Then, when that task is complete, that means that milestone has been reached. At that point, it is
+time to invoice that milestone.
+
+To do that, first return to the sales order -- either via the breadcrumb links, or by navigating to
+:menuselection:`Sales app --> Orders --> Orders` and picking the appropriate sales order.
+
+Back on the sales order form, click the :guilabel:`Milestones` smart button, and check the box in
+the :guilabel:`Reached` column for that particular task.
+
+.. image:: milestone/reached-milestone.png
+   :align: center
+   :alt: What it looks like to mark a milestone as reached via the milestone smart button.
+
+Next, return to the sales order -- either by clicking :guilabel:`View Sales Order` on the
+:guilabel:`Milestones` page, or via the breadcrumb links.
+
+Back on the sales order, the line item for the milestone that's been reached has its
+:guilabel:`Delivered` filled. That's because the milestone has been reached, and therefore
+delivered.
```

Comment:
Not sure if this suggestion is a bit more vague than your previous sentence, but I was slightly confused by the wording "Delivered filled". Hope this rewording is helpful to you! Feel free to cherry-pick your favorite parts from your original and the reworded versions!

```suggestion
Back on the sales order, the milestone line item is marked as :guilabel:`Delivered` filled. That's
because the milestone has been reached.
```

Reply:
Great catch. I forgot to add the word "column" after "Delivered." I made that adjustment, and it reads much smoother now. Thanks again for pointing that out. 👍